iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/ComputeInstanceSshKeysCheck.java

```
@@ -0,0 +1,200 @@
+import org.sonar.iac.terraform.api.tree.ObjectTree;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
+import org.sonar.iac.terraform.symbols.AttributeSymbol;
+import org.sonar.iac.terraform.symbols.ResourceSymbol;
+
+import static org.sonar.iac.common.checks.PropertyUtils.get;
+import static org.sonar.iac.common.checks.PropertyUtils.valueOrNull;
+import static org.sonar.iac.terraform.api.tree.TerraformTree.Kind.BOOLEAN_LITERAL;
+import static org.sonar.iac.terraform.checks.utils.TerraformUtils.attributeAccessToString;
+
+@Rule(key = "S6405")
+public class ComputeInstanceSshKeysCheck extends AbstractNewResourceCheck {
+
+  private static final String MESSAGE = "Make sure that enabling project-wide SSH keys is safe here.";
+  private static final String OMITTING_MESSAGE = "Omitting metadata.block-project-ssh-keys enables project-wide SSH keys. Make sure it is safe here.";
+
+  private final TemplatesCollector collector = new TemplatesCollector("google_compute_instance_template");
+
+  @Override
+  public void initialize(InitContext init) {
+    super.initialize(init);
+    init.register(FileTree.class, (ctx, tree) -> collector.scan(new TreeContext(), tree));
+  }
+
+  @Override
+  protected void registerResourceConsumer() {
+    register("google_compute_instance",
+      resource -> checkMetadata(resource, true));
+
+    register("google_compute_instance_from_template",
+      this::checkInstanceFromTemplate);
+  }
+
+  private void checkInstanceFromTemplate(ResourceSymbol resource) {
+    AttributeSymbol srcTemplate = resource.attribute("source_instance_template");
+    srcTemplate.reportIfAbsent("Missing source_instance_template reference.");
+    if (srcTemplate.isPresent()) {
+      Tree srcTemplateValue = srcTemplate.tree.value();
+      if (!(srcTemplateValue instanceof AttributeAccessTree)) {
+        srcTemplate.ctx.reportIssue(srcTemplateValue, "Template reference of unexpected type.");
```

Why do you think only an AttributeAccess is a valid references type? And even if this would be the case, why we should raise semantic invalid code. It's a rule to detect a specific security hotspot.